### PR TITLE
Added missing items to itemgroup

### LINKED ISSUE #18104: missing firearms in itemgroups
To be thorough, here is a list of all missing firearms from data/json/items/gun/ cross-referenced with data/json/itemgroups/guns.json
  
  /12mm.json         hk_g80
  /223.json             hk_g36
  /300.json             m2010
  /3006.json           survivor_special_700
  /308.json             rifle_308
  /38.json               2_shot_special
  /50.json               m2browning
  /50.json               m2browning_sawn 
  /bio.json              bio_blaster_gun
  /bio.json              bio_laser_gun
  /bio.json              bio_lightning
  /gasoline.json      flamethrower
  /gasoline.json      flamethrower_crude 
  /gasoline.json      flamethrower_simple 
  /gasoline.json      rm451_flamethrower 
  /metal_rail.json    heavy_rail_rifle
  /metal_rail.json    rebar_rifle
  /nail.json             coilgun
  /nail.json             nailgun
  /nail.json             nailrifle
  /signal_flare.json flaregun
  /ups.json            emp_gun
  /ups.json            laser_cannon
  /ups.json            laser_rifle
  /ups.json            unbio_blaster_gun
  /ups.json            v29 
  /ups.json            v29_cheap
  
These weapons are not listed in the /itemgroups/guns.json and I'm sure some weapons are intentionally missing, but some (like hk_g36) may have been overlooked.
  as of ver: 0.C-17711-gaf190e2

### PR DESCRIPTION
**Pull request by BorkBorkGoesTheCode**
Partly fixes #18104.

### CODE REVIEW

**Comment by Reclusive-reptile** [comment]
Spawns for the 5X50 flechette and .300 Win mag also need to be added to itemgroups/ammo.json. They should go into ammo_pistol_milspec and ammo_rifle_milspec respectively with .300 Win mag also added to ammo_rifle_rare at a low weight so the Winchester 70 and Weatherby model 5 can finally have ammo spawn for them.

**Comment by mugling** [comment]
Can you separate out the semantic changes from the formatting ones (separate commits)

**Comment by BorkBorkGoesTheCode** [comment]
If I am holding back any PRs, I'll stop adding more changes to this one.

**Comment by mugling** [comment]
> If I am holding back any PRs, I'll stop adding more changes to this one.

No, the issue is that you're mixing reformatting and semantic changes in the same commit - this makes it harder to see what you're changing.

**Comment by BorkBorkGoesTheCode** [comment]
These are the gameplay changes made in this PR:

guns_rifle_rare
hk_g36: 30

guns_rifle_milspec
m2010: 20

ammo_pistol_milspec
5x50dart: 50
5x50heavy: 30

ammo_pistol_obscure
454_Casull: 100
500_Magnum: 100

ammo_pistol_reloaded
reloaded_454_Casull: 10
reloaded_500_Magnum: 10

ammo_rifle_rare
300_winmag: 40

ammo_rifle_reloaded
reloaded_300_winmag: 15
reloaded_762_m43: 20

File: data/json/itemgroups/ammo.json [hunk 1]
```diff
@@ -31,21 +31,23 @@
...
   {
-    "type" : "item_group",
-    "id" : "ammo_pistol_milspec",
-    "//" : "Military specification pistol ammo found at military sites.",
+    "type": "item_group",
+    "id": "ammo_pistol_milspec",
+    "//": "Military specification pistol ammo found at military sites.",
     "items": [
        [ "46mm", 70 ],
+       [ "5x50dart", 50 ],
```
Comment (mugling) [inline data/json/itemgroups/ammo.json]: Too frequent when compared to 9mm. Consider 20 for the regular and 10 for the heavy.

File: data/json/itemgroups/ammo.json [hunk 2]
```diff
@@ -72,6 +74,8 @@
        [ "reloaded_45_jhp", 100 ],
        [ "reloaded_45_super", 40 ],
        [ "reloaded_46mm", 20 ],
+       [ "reloaded_454_Casull", 10 ],
```
Comment (mugling) [inline data/json/itemgroups/ammo.json]: 5 for both, very rare and valuable

**Comment by Reclusive-reptile** [comment]
Unpopular opinion. The 0.C spawn rates were realistic and balanced. You know that the ammo_rare occurs in one location in the entire game right now? That list contains fmj pistol cartridges that are neither rare in reality or better enough compared to jhp in game to warrant such rarity.

**Comment by mugling** [comment]
> The 0.C spawn rates were realistic and balanced.

`0.C` spawns weren't at all balanced and we had spaghetti JSON

> You know that the ammo_rare occurs in one location in the entire game right now?

Not true. You are aware that we also have C++ mapgen? In any case the subgroups that make it up though are extensively specified

> That list contains fmj pistol cartridges that are neither rare in reality or better enough compared to jhp in game to warrant such rarity.

The issue that FMJ isn't much better than JHP is a significant problem though. Changing item spawns wouldn't affect this.

File: data/json/itemgroups/ammo.json (hunk 1 above)
Comment (mugling) [inline data/json/itemgroups/ammo.json]: Just these outstanding

**Comment by BorkBorkGoesTheCode** [comment]
I have nerfed the remaining cartridges

**Pull request merged**